go-controller/pkg/config/config_test.go

```diff
@@ -299,6 +331,100 @@ var _ = Describe("Config Operations", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 
+	It("uses serviceaccount files", func() {
+		kubeconfigEnvFile, err := createTempFile("kubeconfig.env")
+		Expect(err).NotTo(HaveOccurred())
+		defer os.Remove(kubeconfigEnvFile)
+		os.Setenv("KUBECONFIG", kubeconfigEnvFile)
+		defer os.Setenv("KUBECONFIG", "")
+
+		kubeconfigFile, err := createTempFile("kubeconfig")
+		Expect(err).NotTo(HaveOccurred())
+		defer os.Remove(kubeconfigFile)
+
+		tokenFile, err1 := createTempFileContent("token", "TG9yZW0gaXBzdW0gZ")
+		Expect(err1).NotTo(HaveOccurred())
+		defer os.Remove(tokenFile)
+
+		kubeCAFile, err := createTempFile("ca.crt")
+		Expect(err).NotTo(HaveOccurred())
+		defer os.Remove(kubeCAFile)
+
+		cfgData := fmt.Sprintf(`[kubernetes]
```

Comment:
@pecameron by passing a config file option and having the token/apiserver/etc be the same as the service account files, won't this just test that the config file options are used since they take precedence?

At the lease I think the service account directory stuff should be different than the config file data so we can distinguish the two when checking results later in the test.

Reply:
Changed. I misunderstood how the test worked.

Reply:
@pecameron did you repush yet? Can't see the new changes.

Reply:
Not yet, I'm having trouble getting the ca.crt/token file test to work.

Reply:
@dcbw Fixed, pushed. PTAL